### ISSUE TITLE
FocusAsync not working from blazor desktop wpf

### ISSUE THREAD

**Issue by musictopia2** [dotnet/maui#2528]
When I have code where after it first renders it will focusasync a control, its not working at all.  No runtime error.  But it also does not focus either.  If there is a workaround, would like to know what the workaround is because its very common to need to focus on a particular element when the page first loads.

**Comment by Eilon** [comment]
Hi @musictopia2 , I was able to get this to work, but it's a bit tricky because there are two "levels" of focus at play here:

1. Which WPF (or WinForms, or whatever) control has focus (in our case, the BlazorWebView control).
2. _Within_ that BlazorWebView control, which HTML element has focus

In order for the user to see the HTML element being focused, both (1) and (2) must be done.

Here's how I did it:

For (1) I added this code to my WPF Blazor Desktop app's MainWindow constructor to make sure the BlazorWebView's inner WebView2 is in focus:

```c#
public MainWindow()
{
	...
	InitializeComponent();
	...

	this.Loaded += (_, __) => blazorWebView1.WebView.Focus();
}
```

And for (2) within my Razor file I did this to focus a particular HTML element within the BlazorWebView:

```razor
    <InputText id="name" @bind-Value="exampleModel.Name" @ref="textInput" />

@code
{
    private InputText textInput;

    protected override async Task OnAfterRenderAsync(bool firstRender)
    {
        if (firstRender)
        {
            await textInput.Element.Value.FocusAsync();
        }
    }
}
```

Of course, the exact details of each app will be a bit different, but those should be the key parts to get working.
